### PR TITLE
Short URLs

### PR DESCRIPTION
**Pull request by markspolakovs**
Goes in tandem with https://github.com/UniversityRadioYork/2016-site/pull/297

### CODE REVIEW

File: schema/patches/10.sql [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+CREATE TABLE public.short_url_clicks (
+    click_id BIGSERIAL PRIMARY KEY,
+    short_url_id INTEGER REFERENCES public.short_urls (short_url_id) ON DELETE CASCADE,
+    click_time TIMESTAMPTZ NOT NULL,
+    user_agent TEXT DEFAULT NULL,
+    ip_address INET DEFAULT NULL
+);
+
+INSERT INTO public.l_action (descr, phpconstant) VALUES ('Edit Short URLs', 'AUTH_EDITSHORTURLS');
```
Comment (michael-grace) [inline schema/patches/10.sql:17]: could this be added in `schema/whichever` file so its with all the others, and all of those get added when someone makes a local myradio. and we can just add this one ourselves to prod

File: src/Classes/ServiceAPI/MyRadio_ShortURL.php [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+            [
+                'title' => 'Edit Short URL'
+            ]
+        ))->addField(
+            new MyRadioFormField(
+                'slug',
+                MyRadioFormField::TYPE_TEXT,
+                [
+                    'label' => 'Slug',
+                    'explanation' => "The bit that goes after https://$domain. Don't include $domain or a slash."
```
Comment (michael-grace) [inline src/Classes/ServiceAPI/MyRadio_ShortURL.php:143]: does this/could this fill from the config `short_url` or whatever the fields called?

Comment (markspolakovs) [reply src/Classes/ServiceAPI/MyRadio_ShortURL.php:143]: I don't think we have one of those - might you be confusing it with `short_name`?

File: schema/patches/10.sql (hunk 1 above)
Comment (markspolakovs) [reply schema/patches/10.sql:17]: Oh yeah that exists, will change

File: src/Classes/ServiceAPI/MyRadio_ShortURL.php (hunk 2 above)
Comment (michael-grace) [reply src/Classes/ServiceAPI/MyRadio_ShortURL.php:143]: uh.. we've got `base_url` but thats got the /myradio bit on.

**Review by michael-grace — APPROVED** [review]

**Pull request merged**